File: scripts/statusline.js
#!/usr/bin/env node
const fs = require("fs");
const path = require("path");

const ROLE_TITLE_OVERRIDES = {
  qa: "QA",
};

const PROVIDER_TITLE_OVERRIDES = {
  claude: "Claude",
  codex: "Codex",
  gemini: "Gemini",
};

const titleize = (name, overrides) => {
  if (overrides && overrides[name]) return overrides[name];
  return name
    .split(/[-_]+/)
    .filter(Boolean)
    .map((part) => part[0].toUpperCase() + part.slice(1))
    .join(" ");
};

const projectDir = process.env.CLAUDE_PROJECT_DIR || process.cwd();
const settingsPath = path.join(projectDir, ".claude", "settings.json");

let settings = {};
try {
  settings = JSON.parse(fs.readFileSync(settingsPath, "utf8"));
} catch (err) {
  settings = {};
}

const config = settings.aiArchitect || {};
const roles = config.roles || {};

const entries = Object.entries(roles)
  .filter(([, cfg]) => cfg && cfg.enabled !== false)
  .map(([role, cfg]) => {
    const provider = (cfg && cfg.provider) || "claude";
    const roleTitle = titleize(role, ROLE_TITLE_OVERRIDES);
    const providerTitle = titleize(provider, PROVIDER_TITLE_OVERRIDES);
    return `${roleTitle} [${providerTitle}]`;
  });

process.stdout.write(entries.join(" | "));

File: scripts/statusline.py
#!/usr/bin/env python3
import re
from ai_lib import load_config, project_root

ROLE_TITLE_OVERRIDES = {
    "qa": "QA",
}
PROVIDER_TITLE_OVERRIDES = {
    "claude": "Claude",
    "codex": "Codex",
    "gemini": "Gemini",
}


def _titleize(name: str, overrides: dict) -> str:
    if name in overrides:
        return overrides[name]
    parts = re.split(r"[-_]+", name)
    return " ".join(part.capitalize() for part in parts if part)


def main() -> int:
    root = project_root()
    config = load_config(root)
    roles = config.get("roles", {})

    entries = []
    for role_name, role_cfg in roles.items():
        if not role_cfg.get("enabled", True):
            continue
        provider = role_cfg.get("provider", "claude")
        role_title = _titleize(role_name, ROLE_TITLE_OVERRIDES)
        provider_title = _titleize(provider, PROVIDER_TITLE_OVERRIDES)
        entries.append(f"{role_title} [{provider_title}]")

    if entries:
        print(" | ".join(entries))
    else:
        print("")
    return 0


if __name__ == "__main__":
    raise SystemExit(main())
